### PR TITLE
add gedcomx simplifier

### LINKED ISSUE #23: GedcomX simplifier functions
FamilySearch represents genealogical data in a format called GedcomX. It is very verbose. We want the LLM to receive and send a data in a "simplified" GedcomX format. 

We need two functions (in typescript). The MCP servers will call these functions as needed to convert between the two formats:
1. GedcomX (from tool) -> SimplifiedGedcomX (for LLM)
2. SimplifiedGedcomX (from LLM) -> GedcomX (for tool)

Here are the simplification rules:

- URI prefixes dropped: Birth not http://gedcomx.org/Birth. Saves tokens, costs nothing in clarity.
- nameForms and parts collapsed: given and surname directly on the name. The deeper nesting only matters for multi-script names you probably won’t encounter.
- preferred on names and primary on facts as explicit booleans. The redundancy with first-position convention is intentional — both signals point the same way, so the model is less likely to miss it.
- parent/child instead of person1/person2 for ParentChild. The semantic role is clearer than the positional naming.
- Couple keeps person1/person2 because the relationship is symmetric.
- quality field on sources, mimicking GEDCOM’s QUAY. This is the single most useful piece of metadata for discrepancy reasoning, and I’d encourage you to populate it even if your underlying GedcomX doesn’t have it natively.
- IDs on names and facts, not just persons. Lets the LLM (or your application) refer to “fact F1” or “name N2" unambiguously when generating research recommendations.

And an example:

GedcomX

```
{
  "persons": [
    {
      "id": "I1",
      "gender": { "type": "http://gedcomx.org/Male" },
      "names": [
        {
          "id": "N1",
          "preferred": true,
          "type": "http://gedcomx.org/BirthName",
          "nameForms": [
            {
              "fullText": "John Doe",
              "parts": [
                { "type": "http://gedcomx.org/Given", "value": "John" },
                { "type": "http://gedcomx.org/Surname", "value": "Doe" }
              ]
            }
          ],
          "sources": [
            {
              "description": "#S1",
              "qualifiers": [
                { "name": "http://gedcomx.org/CitationDetail", "value": "1920 Census, Denver, ED 47" }
              ]
            }
          ]
        },
        {
          "id": "N2",
          "type": "http://gedcomx.org/AlsoKnownAs",
          "nameForms": [
            {
              "fullText": "Jonathan Doe",
              "parts": [
                { "type": "http://gedcomx.org/Given", "value": "Jonathan" },
                { "type": "http://gedcomx.org/Surname", "value": "Doe" }
              ]
            }
          ],
          "sources": [
            {
              "description": "#S2",
              "qualifiers": [
                { "name": "http://gedcomx.org/CitationDetail", "value": "Death certificate no. 4521" }
              ]
            }
          ]
        }
      ],
      "facts": [
        {
          "id": "F1",
          "type": "http://gedcomx.org/Birth",
          "date": { "original": "1900", "formal": "+1900" },
          "place": { "original": "Denver, Colorado, USA" },
          "sources": [
            {
              "description": "#S1",
              "qualifiers": [
                { "name": "http://gedcomx.org/CitationDetail", "value": "1920 Census, Denver, ED 47" }
              ]
            }
          ]
        }
      ]
    },
    {
      "id": "I2",
      "gender": { "type": "http://gedcomx.org/Male" },
      "names": [
        {
          "preferred": true,
          "nameForms": [
            {
              "fullText": "Frank Doe",
              "parts": [
                { "type": "http://gedcomx.org/Given", "value": "Frank" },
                { "type": "http://gedcomx.org/Surname", "value": "Doe" }
              ]
            }
          ]
        }
      ]
    },
    {
      "id": "I3",
      "gender": { "type": "http://gedcomx.org/Female" },
      "names": [
        {
          "preferred": true,
          "nameForms": [
            {
              "fullText": "Sally Smith",
              "parts": [
                { "type": "http://gedcomx.org/Given", "value": "Sally" },
                { "type": "http://gedcomx.org/Surname", "value": "Smith" }
              ]
            }
          ]
        }
      ]
    }
  ],
  "relationships": [
    {
      "id": "R1",
      "type": "http://gedcomx.org/ParentChild",
      "person1": { "resource": "#I2" },
      "person2": { "resource": "#I1" },
      "sources": [
        {
          "description": "#S1",
          "qualifiers": [
            { "name": "http://gedcomx.org/CitationDetail", "value": "1920 Census, Denver, ED 47" }
          ]
        }
      ]
    },
    {
      "id": "R2",
      "type": "http://gedcomx.org/ParentChild",
      "person1": { "resource": "#I3" },
      "person2": { "resource": "#I1" },
      "sources": [
        {
          "description": "#S1",
          "qualifiers": [
            { "name": "http://gedcomx.org/CitationDetail", "value": "1920 Census, Denver, ED 47" }
          ]
        }
      ]
    },
    {
      "id": "R3",
      "type": "http://gedcomx.org/Couple",
      "person1": { "resource": "#I2" },
      "person2": { "resource": "#I3" },
      "facts": [
        {
          "type": "http://gedcomx.org/Marriage",
          "date": { "original": "1898", "formal": "+1898" },
          "place": { "original": "Denver, Colorado, USA" },
          "sources": [
            {
              "description": "#S3",
              "qualifiers": [
                { "name": "http://gedcomx.org/CitationDetail", "value": "Marriage license, Denver County" }
              ]
            }
          ]
        }
      ]
    }
  ],
  "sourceDescriptions": [
    {
      "id": "S1",
      "citations": [
        { "value": "1910 United States Federal Census. National Archives and Records Administration." }
      ],
      "titles": [{ "value": "1910 U.S. Federal Census" }]
    },
    {
      "id": "S2",
      "citations": [
        { "value": "Utah Death Certificates, 1904-1965. Utah State Archives." }
      ],
      "titles": [{ "value": "Utah Death Certificates" }]
    },
    {
      "id": "S3",
      "citations": [
        { "value": "Denver County Marriage Records. Colorado State Archives." }
      ],
      "titles": [{ "value": "Denver County Marriage Records" }]
    }
  ]
}
```

Simplified

```
{
  "persons": [
    {
      "id": "I1",
      "gender": "Male",
      "names": [
        {
          "id": "N1",
          "preferred": true,
          "given": "John",
          "surname": "Doe",
          "type": "BirthName",
          "sources": [{ "ref": "S1", "page": "...", "quality": 2 }]
        },
        {
          "id": "N2",
          "given": "Jonathan",
          "surname": "Doe",
          "type": "AlsoKnownAs",
          "sources": [{ "ref": "S2", "page": "...", "quality": 3 }]
        }
      ],
      "facts": [
        {
          "id": "F1",
          "type": "Birth",
          "primary": true,
          "date": "1900",
          "place": "Denver, Colorado, USA",
          "sources": [{ "ref": "S1", "page": "...", "quality": 2 }]
        }
      ]
    }
  ],
  "relationships": [
    {
      "id": "R1",
      "type": "ParentChild",
      "parent": "I2",
      "child": "I1",
      "sources": [{ "ref": "S1", "page": "...", "quality": 2 }]
    },
    {
      "id": "R2",
      "type": "Couple",
      "person1": "I2",
      "person2": "I3",
      "facts": [
        {
          "type": "Marriage",
          "date": "1898",
          "place": "Denver, Colorado, USA",
          "sources": [{ "ref": "S3", "page": "...", "quality": 3 }]
        }
      ],
      "sources": [{ "ref": "S1", "page": "...", "quality": 2 }]
    }
  ],
  "sources": [
    { "id": "S1", "title": "1910 U.S. Federal Census", "author": "U.S. Census Bureau" },
    { "id": "S2", "title": "Utah Death Certificates", "author": "Utah State Archives" }
  ]
}
```

### PR DESCRIPTION
**Pull request by Gennecis**
Closes #23.

  ## Summary

  - Adds `docs/specs/gedcomx-convert-spec.md` — implementation spec for two
    pure utility functions that round-trip between FamilySearch's full
    GedcomX format and the simplified format defined in
    `simplified-gedcomx-spec.md`.
  - Implements `toSimplified()` and `toGedcomX()` in
    `mcp-server/src/utils/gedcomx-convert.ts` (new `src/utils/` directory —
    first non-tool/non-auth shared module; intended consumers are the
    upcoming `tree` and `cets` tools).
  - 25 vitest cases in `tests/utils/gedcomx-convert.test.ts` cover all 14
    transformation rules, edge cases, and the omit-when-false schema
    compliance bar.

  ## Test plan

  - [x] `cd mcp-server && npm run build` — clean
  - [x] `cd mcp-server && npm test` — 160/160 passing (25 new, no regressions)
  - [x] `cd mcp-server && npx tsx dev/try-gedcomx-convert.ts` — Turner-family
        smoke output matches the spec's "Expected output" section
  - [x] Spec-review: `docs/specs/gedcomx-convert-spec.md` aligns with the
        schema authority in `docs/specs/simplified-gedcomx-spec.md`

### CODE REVIEW

File: docs/specs/gedcomx-convert-spec.md [hunk 1]
```diff
@@ -0,0 +1,717 @@
...
+  names?: SimplifiedName[];
+  facts?: SimplifiedFact[];
+  sources?: SimplifiedSourceReference[];
+};
+
+export type SimplifiedName = {
+  id?: string;
+  type?: string;             // PascalCase, e.g. "BirthName"
+  preferred?: boolean;       // Present only when true
+  given?: string;
```
Comment (DallanQ) [inline docs/specs/gedcomx-convert-spec.md:144]: I think we need to add prefix and suffix in addition to given and surname

File: docs/specs/gedcomx-convert-spec.md [hunk 2]
```diff
@@ -0,0 +1,717 @@
...
+```
+
+`toGedcomX`: re-wrap as `{ type: "http://gedcomx.org/<value>" }`. For
+`"Unknown"`, omit the `gender` field entirely.
+
+### 3. Names — given / surname extraction
+
+**Primary path:** read `nameForms[0].parts`, taking the first `Given`-typed
+entry and the first `Surname`-typed entry.
+
```
Comment (DallanQ) [inline docs/specs/gedcomx-convert-spec.md:233]: Also handle Prefix and Suffix. Warn on anything other than Given, Surname, Prefix, or Suffix so we can see if we need to handle anything else.

File: docs/specs/gedcomx-convert-spec.md [hunk 3]
```diff
@@ -0,0 +1,717 @@
...
+**Fallback path:** if `parts` is missing or empty but `nameForms[0].fullText`
+is present, split on the **last** whitespace — everything before becomes
+`given`, the trailing token becomes `surname`. A single token (mononym)
+becomes `surname` with `given: ""`.
+
```
Comment (DallanQ) [inline docs/specs/gedcomx-convert-spec.md:246]: Hopefully we don't have to do this very often. Splitting on the last whitespace doesn't work for latin double-surnames.

File: docs/specs/gedcomx-convert-spec.md [hunk 4]
```diff
@@ -0,0 +1,717 @@
...
+- `names[0]` → `preferred: true`
+- `names[1..n]` → field **omitted**
+
+`toGedcomX`: place the `preferred: true` name first; never emit
+`preferred: false`.
+
+### 5. `primary` on facts
+
+The first fact in `facts[]` is the primary fact.
+
```
Comment (DallanQ) [inline docs/specs/gedcomx-convert-spec.md:285]: not true. Only set primary in simplfied gedcomx when primary is set in gedcomx and vice-versa
I can have a birth, marriage, death fact and they can all be primary. But if I have *two* birth facts, one can be primary and the other not primary.

File: docs/specs/gedcomx-convert-spec.md [hunk 5]
```diff
@@ -0,0 +1,717 @@
...
+```
+
+`toGedcomX`: reconstruct `nameForms: [{ fullText, parts }]`:
+- `fullText` is `` `${given} ${surname}`.trim() ``
+- `parts` contains entries only for non-empty `given` / `surname`
+
+### 4. `preferred` on names
+
+The first name in `names[]` is the preferred name.
+
```
Comment (DallanQ) [inline docs/specs/gedcomx-convert-spec.md:265]: Only set preferred in simplfied gedcomx when preferred is set in gedcomx and vice-versa

File: docs/specs/gedcomx-convert-spec.md [hunk 6]
```diff
@@ -0,0 +1,717 @@
...
+↓
+{ "ref": "S1", "page": "1920 Census, ED 47", "quality": 3 }
+```
+
+- `description: "#S1"` → `ref: "S1"` (strip `#`)
+- Qualifier with `name === "http://gedcomx.org/CitationDetail"` → `page`
+- Qualifier with `name === "fsmcp:quality"` → `quality` as `Number(value)`
+- All other qualifiers are dropped
+- When `quality` is absent or non-numeric, **omit** the field
+
```
Comment (DallanQ) [inline docs/specs/gedcomx-convert-spec.md:364]: If raw fsmcp:quality can be any string, not just a number, then let's allow quality to be any string as well.

File: docs/specs/gedcomx-convert-spec.md [hunk 7]
```diff
@@ -0,0 +1,717 @@
...
+  "citation": "1910 United States Federal Census. NARA.",
+  "url": "https://www.archives.gov/..."
+}
+```
+
+- `titles[0].value` → `title` (omit if titles missing/empty)
+- `citations[0].value` → `citation` (omit if citations missing/empty)
+- `about` → `url`
+- `author` has no GedcomX equivalent — `toSimplified` leaves it undefined.
+  `toGedcomX` drops `author` (no destination field).
```
Comment (DallanQ) [inline docs/specs/gedcomx-convert-spec.md]: Let's omit author from simplified gedcomx then.

File: docs/specs/gedcomx-convert-spec.md [hunk 8]
```diff
@@ -0,0 +1,717 @@
...
+  "latitude": 39.7817,
+  "longitude": -89.6501
+}
+```
+
+- `names[0].value` → `name` (omit if names missing/empty)
+- Top-level array key stays `places` on both sides
+
+### 13. `Census` fact type
+
```
Comment (DallanQ) [inline docs/specs/gedcomx-convert-spec.md]: Let's omit Census and always use Residence.

**Review by DallanQ — CHANGES_REQUESTED** [review]
Also, at least some unit tests should round-trip from raw gedcomx to simplified gedcomx and back again and make sure the raw gedcomx is the same. And some unit tests should round-trip from simplified gedcomx to raw gedcomx and back again and make sure the simplified gedcomx is the same.
